File: problems1-10/prob9.py
# -*- coding: utf-8 -*-
"""
Created on Thu May 11 00:09:04 2017

@author: Matt
"""

"""
A Pythagorean triplet is a set of three natural numbers, a < b < c, for which,

a2 + b2 = c2
For example, 32 + 42 = 9 + 16 = 25 = 52.

There exists exactly one Pythagorean triplet for which a + b + c = 1000.
Find the product abc.
"""
import math


def findThatTriple(target): #strat is guess and check brute force
    for a in range(1,500): #this is already bad because I'm checking the same pair twice, change to a while
        
        for b in range(1,500):
            hyp=a**2+b**2
            c=math.sqrt(hyp)
            if c%1==0:
                if a+b+c==target:
                    return (a,b,c)
            
            
if __name__=="__main__":
    print(findThatTriple(1000))

File: problems1-10/prob9.cpp
using namespace std;

#include <iostream>
#include <cmath>

string findThatTriple(int target);

int main(){
  cout << findThatTriple(1000) << endl;
  return 0;
}

string findThatTriple(int target){
  string triple="";
  for(int a = 1; a<500;a++){
    for(int b = 1; b < 500; b++){
      int hyp = pow(a,2)+pow(b,2);
      double c = sqrt(hyp);
      if(fmod(c,1)==0){
	if(a+b+c==target){
	  triple=to_string(a)+" "+to_string(b)+" "+to_string(c);
	}
      }
    }
  }

  return triple;
}
